prepare/metrics/rag.py

```diff
@@ -416,3 +416,138 @@
     add_to_catalog(
         metric, f"metrics.rag.response_generation.{axis}.{base_metric}", overwrite=True
     )
+
+# end to end
```

Comment:
i think it would be nice to have any new answer correctness and context correctness metrics in these modules:
https://github.com/IBM/unitxt/blob/main/prepare/metrics/rag_answer_correctness.py
https://github.com/IBM/unitxt/blob/main/prepare/metrics/rag_context_correctness.py

splitting up the metrics across modules makes things easier to follow, and allows reuse of the testing code that exists in the above modules.

Reply:
I have to admit that I am lost with the numerous number of RAG metrics...

Reply:
i think you are not the only one .. for that reason i started to split the metrics into separate modules (linked above) according to their type ..